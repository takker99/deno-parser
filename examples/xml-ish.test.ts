import { XML } from "./xml-ish.ts";
import { assertSnapshot } from "@std/testing/snapshot";
import { assertEquals } from "@std/assert";

Deno.test("xml basic", () => {
  assertEquals(XML.tryParse(`<a key="val" />`), {
    name: "a",
    attributes: { key: "val" },
    children: [],
  });
  assertEquals(XML.tryParse(`<a key="val">foo<b /></a>`), {
    name: "a",
    attributes: { key: "val" },
    children: ["foo", { name: "b", attributes: {}, children: [] }],
  });
});

Deno.test("xml large example", (t) =>
  assertSnapshot(
    t,
    XML.parse(
      `
    <Map>
      <Entry key="string-a"><String>alpha</String></Entry>
      <Entry key="string-b"><String>bravo</String></Entry>
      <Entry key="string-c"><String>charlie</String></Entry>
      <Entry key="list">
        <List>
          <String>a</String>
          <String>b</String>
          <String>c</String>
        </List>
      </Entry>
      <Entry key="null"><Null /></Entry>
      <Entry key="empty" />
    </Map>
    `,
    ),
  ));
