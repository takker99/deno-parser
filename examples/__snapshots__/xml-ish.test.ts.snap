export const snapshot = {};

snapshot[`xml large example 1`] = `
{
  type: "ParseOK",
  value: {
    attributes: {},
    children: [
      "
      ",
      {
        attributes: {
          key: "string-a",
        },
        children: [
          {
            attributes: {},
            children: [
              "alpha",
            ],
            name: "String",
          },
        ],
        name: "Entry",
      },
      {
        attributes: {
          key: "string-b",
        },
        children: [
          {
            attributes: {},
            children: [
              "bravo",
            ],
            name: "String",
          },
        ],
        name: "Entry",
      },
      {
        attributes: {
          key: "string-c",
        },
        children: [
          {
            attributes: {},
            children: [
              "charlie",
            ],
            name: "String",
          },
        ],
        name: "Entry",
      },
      {
        attributes: {
          key: "list",
        },
        children: [
          "
        ",
          {
            attributes: {},
            children: [
              "
          ",
              {
                attributes: {},
                children: [
                  "a",
                ],
                name: "String",
              },
              {
                attributes: {},
                children: [
                  "b",
                ],
                name: "String",
              },
              {
                attributes: {},
                children: [
                  "c",
                ],
                name: "String",
              },
            ],
            name: "List",
          },
        ],
        name: "Entry",
      },
      {
        attributes: {
          key: "null",
        },
        children: [
          {
            attributes: {},
            children: [],
            name: "Null",
          },
        ],
        name: "Entry",
      },
      {
        attributes: {
          key: "empty",
        },
        children: [],
        name: "Entry",
      },
    ],
    name: "Map",
  },
}
`;
